sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c

```diff
@@ -562,8 +562,21 @@ inline static size_t iphc_nhc_udp_encode(gnrc_pktsnip_t *udp, ipv6_hdr_t *ipv6_h
 
     /* Set UDP header ID (rfc6282#section-5). */
     ipv6_hdr->nh |= NHC_UDP_ID;
-    /* shrink udp allocation to final size */
-    gnrc_pktbuf_realloc_data(udp, nhc_len);
+
+    if (udp->type == GNRC_NETTYPE_IPV6) {
+        /* forwarded ipv6 packet */
+        size_t diff = sizeof(udp_hdr_t) - nhc_len;
+        for (int i = nhc_len; i < (udp->size - diff); i++) {
```

Comment:
Apparently this is the line which is breaking Murdock, but it's strange that Murdock didn't show the error on this check... How can we proceed to this? should I only change the the type of `int i` to `size_t i` to make Murdock happy? Is this the real cause of Murdock failing everywhere?